src/components/Select/Select.jsx

```diff
@@ -29,24 +29,38 @@ export default class Select extends React.Component {
     document.addEventListener('click', this.closePopover, true);
 
     // catch the keypress to move the selected items up or down
-    document.addEventListener('keydown', this.keyDownPressed, true);
+    this.selectNode && this.selectNode.addEventListener('keydown', this.keyDownPressed);
   }
 
   componentWillUnmount() {
     document.removeEventListener('click', this.closePopover, true);
-    document.removeEventListener('keydown', this.keyDownPressed, true);
+    this.selectNode && this.selectNode.removeEventListener('keydown', this.keyDownPressed);
```

Comment:
I haven't seen this before 😊 would you mind teaching me why we need `this.selectNode &&` ?

Reply:
I can help here @gomezjuliana! 

The line here is functionally equivalent to:

```js
if (this.selectNode) {
  this.selectNode.removeEventListener('keydown', this.keyDownPressed);
}
```

The && operator evaluates the first part of the expression first, and if it's falsy (i.e., null, undefined, etc.) it stops right there. This is also why you'll sometimes see this pattern in React/JSX:

```jsx
<div>
  {shouldShowThing && <TheThing />}
</div>
```

Which only renders `<TheThing />` when `shouldShowThing` is truthy. 😄

Let me know if that makes sense!

Reply:
Makes sense! Thanks @hamstu 🌮 !!